src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -4,11 +4,16 @@
 import htsjdk.samtools.CigarOperator;
 import org.broadinstitute.hellbender.utils.BaseUtils;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.locusiterator.AlignmentStateMachine;
 import org.broadinstitute.hellbender.utils.read.ReadUtils;
 
 import java.util.*;
 
-public final class PileupElement implements Comparable<PileupElement> {
+/**
+ * Represents a position in the read pileup.
+ */
+public final class PileupElement {
+    //FIXME: this static linked list is modifiable
```

Comment:
the API wants a LinkedList (massive pain but it does because it wants something that is both a List and a Deque, and LinkedList is one such thing) and there's no `Collections.emptyLinkedList()`